cloudinit/util.py

```diff
@@ -1739,7 +1747,8 @@ def get_output_cfg(cfg, mode):
     for i in range(len(ret)):
         if not ret[i]:
             continue
-        val = ret[i].lstrip()
+        # str because the previous conditional continue
+        val = cast(str, ret[i]).lstrip()
```

Comment:
What about changing the for loop to be something like
`if index, item in enumerate(ret):`

Mypy will understand those semantics better and not complain after seeing a `if not item`. It's also a little easier to read (imo), and then the only place the index needs to be used in the last line of the loop with `ret[index] = val`.

Reply:
Good point, with this, it is much more clear.